### PR TITLE
fix: wait for pidfile

### PR DESCRIPTION
**Pull request by yitsushi**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by yitsushi** [comment]
e2e run: https://github.com/weaveworks/flintlock/actions/runs/1520405713

Failed, but unrelated:
```
time="2021-11-30T10:39:08Z" level=error msg="failed to reconcile vmid ns0/mvm0: executing plan: executing plan steps: executing steps: executing step runtime_volume_mount: mount images docker.io/richardcase/ubuntu-bionic-test:cloudimage_v0.0.1 for volume use: getting image docker.io/richardcase/ubuntu-bionic-test:cloudimage_v0.0.1 for owner ns0/mvm0: pulling image using containerd: failed to copy: httpReadSeeker: failed open: unexpected status code https://registry-1.docker.io/v2/richardcase/ubuntu-bionic-test/manifests/sha256:371792335fe2360e0fd30108e42eb565c990140aeb9bcc796de10a59da54beb6: 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit" controller=microvm
```

```
429 Too Many Requests - Server message: toomanyrequests:
  You have reached your pull rate limit.
  You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

**Comment by yitsushi** [comment]
New run: https://github.com/weaveworks/flintlock/actions/runs/1520405713

**Comment by yitsushi** [comment]
New run hope it will not hit rate limit now: https://github.com/weaveworks/flintlock/runs/4366216033?check_suite_focus=true

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ func TestE2E(t *testing.T) {
 
 	log.Println("TEST STEP: getting (and verifying) existing MicroVM")
 	Eventually(func(g Gomega) error {
+		g.Expect(fmt.Sprintf(fcPath, mvmNS, secondMvmID) + "/firecracker.pid").To(BeAnExistingFile())
```
Comment (Callisto13) [inline test/e2e/e2e_test.go:52]:
```suggestion
		g.Expect(fmt.Sprintf(fcPath, mvmNS, mvmID) + "/firecracker.pid").To(BeAnExistingFile())
```

**Comment by yitsushi** [comment]
Let's try again: https://github.com/weaveworks/flintlock/actions/runs/1521365501

**Comment by Callisto13** [comment]
10th time's the charm :trollface:

**Review by Callisto13 — APPROVED** [review]

**Pull request merged**